src/NServiceBus.RabbitMQ/MessagePump.cs

```diff
@@ -157,14 +158,18 @@ public void Start(PushRuntimeSettings limitations)
                     poisonMessageForwarder.ForwardPoisonMessageToErrorQueue(message, ex, settings.ErrorQueue);
                 }
 
+                CancellationTokenSource tokenSource = null;
                 if (pushMessage)
                 {
-                    await PushMessageToPipe(messageId, headers, new MemoryStream(message.Body ?? new byte[0])).ConfigureAwait(false);
+                    tokenSource = new CancellationTokenSource();
+                    await PushMessageToPipe(messageId, headers, tokenSource, new MemoryStream(message.Body ?? new byte[0])).ConfigureAwait(false);
                 }
 
-                if (!noAck)
+                var token = (tokenSource?.Token).GetValueOrDefault();
+
+                if (!noAck && !token.IsCancellationRequested)
```

Comment:
@Scooletz wouldn't it make more sense to just check if `token.IsCancellationRequested == true` and then call `channel.BasicReject` ?